x-pack/plugin/core/src/main/java/org/elasticsearch/datastreams/logsdb/PatternMatcher.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.datastreams.logsdb;
+
+import java.util.regex.Pattern;
+
+public class PatternMatcher {
```

Comment:
I don't think this is needed. We already have logic that deals with pattern matches for index templates. I think just using is good enough: `Regex.simpleMatch(logs-*-*, dataStreamName`)`